govuk_chat_evaluation/rag_answers/evaluate.py

```diff
@@ -1,31 +1,227 @@
+import logging
+import os
 from pathlib import Path
+from typing import cast, Any
+from dataclasses import asdict
+from functools import cached_property
+from pydantic import BaseModel
+import pandas as pd
 
 from deepeval import evaluate as deepeval_evaluate
-from deepeval.metrics import GEval
-from deepeval.test_case import LLMTestCaseParams
+from deepeval.evaluate import TestResult
+from deepeval.test_case import LLMTestCase, MLLMTestCase
+from deepeval.metrics import BaseMetric
 
+
+from ..timing import Timer
 from ..file_system import jsonl_to_models
-from .data_models import EvaluationTestCase
+from .data_models import EvaluationTestCase, EvaluationConfig
+
+
+logger = logging.getLogger(__name__)
 
 
 # would expect we need to pass config object through if that has metrics configuration
-def evaluate_and_output_results(_output_dir: Path, evaluation_data_path: Path):
-    models = jsonl_to_models(evaluation_data_path, EvaluationTestCase)
-    run_deepeval_evaluation(models)
+def evaluate_and_output_results(
+    _output_dir: Path, 
+    evaluation_data_path: Path, 
+    evaluation_config: EvaluationConfig, 
+    deepeval_evaluate_params: dict
```

Comment:
I think rather than passing these through a few levels we'd do better to hardcode them initially as close to deepeval execute as we can and then if they become something we want to change have them be properties of the config object this accepts

Reply:
Yeah sorry meant to comment on just that one line. 

Yeah I'd suggest getting them just hardcoded really close to where they're used and tweaking them directly. Otherwise it's a lot of passing of somewhat arbitrary data through the system.